src/codewhisperer/activation.ts

```diff
@@ -259,49 +263,49 @@ export async function activate(context: ExtContext): Promise<void> {
                 await RecommendationHandler.instance.onCursorChange(e)
             }),
             vscode.workspace.onDidChangeTextDocument(async e => {
+                const editor = vscode.window.activeTextEditor
+                if (!editor) {
+                    return
+                }
+                if (e.document !== editor.document) {
+                    return
+                }
+                if (!runtimeLanguageContext.isLanguageSupported(e.document.languageId)) {
+                    return
+                }
+
                 /**
                  * CodeWhisperer security panel dynamic handling
                  */
-                if (e.document === vscode.window.activeTextEditor?.document) {
-                    disposeSecurityDiagnostic(e)
-                }
+                disposeSecurityDiagnostic(e)
```

Comment:
is the change on purpose? (removing the `if` check), not sure what this line is doing, just want do double check this change is intentional

Reply:
The check is moved to above lines